### PR TITLE
Added test case for 3scale custom policy

### PR DESCRIPTION
**Pull request by trepel**
# Description

Added a new test case, see https://issues.redhat.com/browse/MGDAPI-370. Hopefully we won't need to do this too often.

# Verification Steps
eye review only I guess. I had a few calls with 3scale QE and the comments in MGDAPI-370 were verified against RHOAM with 3scale 2.9.1.
But you can try yourself for sure.

### CODE REVIEW

File: test-cases/tests/products/h24-verify-selfmanaged-apicast-and-custom-policy.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+with
+
+```
+    strategy:
+      type: Source
+      sourceStrategy:
+        from:
+          kind: ImageStreamTag
+          namespace: selfmanaged-apicast
+          name: apicast-gateway-rhel8:3scale2.9.1
```
Comment (psturc) [inline test-cases/tests/products/h24-verify-selfmanaged-apicast-and-custom-policy.md:66]:
```suggestion
          name: apicast-gateway-rhel8:3scale<version>
```
would it be also worth mentioning how to find out the 3scale version?

File: test-cases/tests/products/h24-verify-selfmanaged-apicast-and-custom-policy.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+_9. Start custom policy builds_
+
+- `oc start-build apicast-example-policy`
+- `oc start-build apicast-custom-policies`
+- once finished, the `image-registry.openshift-image-registry.svc:5000/selfmanaged-apicast/apicast-policy:example` should be available to be used as image for self-managed APIcast
+
+_10. Install "Red Hat Integration - 3scale APIcast gateway" operator via Operator Hub in OSD web console for the namespace_
+
+- do not sue community version!
```
Comment (psturc) [inline test-cases/tests/products/h24-verify-selfmanaged-apicast-and-custom-policy.md:141]:
```suggestion
- do not use community version!
```

**Review by psturc — COMMENTED** [review]
just a few comments/suggestions, otherwise LGTM

File: test-cases/tests/products/h24-verify-selfmanaged-apicast-and-custom-policy.md (hunk 2 above)
Comment (trepel) [reply test-cases/tests/products/h24-verify-selfmanaged-apicast-and-custom-policy.md:141]: good catch, will fix it

File: test-cases/tests/products/h24-verify-selfmanaged-apicast-and-custom-policy.md (hunk 1 above)
Comment (trepel) [reply test-cases/tests/products/h24-verify-selfmanaged-apicast-and-custom-policy.md:66]: apparently, this is not clear enough. that whole `3scale<version>` is an image tag. So you need to look in the catalog and see what tags are available and pick one - likely the one that corresponds with the 3scale version in RHOAM. Will improve the wording.

**Comment by trepel** [comment]
@psturc Your comments have been addressed, would you mind taking another look?

I added two bullet points to the step 5 about the tag so it is more clear hopefully.

**Pull request merged**